utils/logger.go

```diff
@@ -0,0 +1,109 @@
+// this is a new logger interface for mattermost
+
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"runtime"
+
+	l4g "github.com/alecthomas/log4go"
+)
+
+// this pattern allows us to "mock" the underlying l4g code when unit testing
+var debug = l4g.Debug
```

Comment:
I wish I could take credit for the idea. I found this approach on StackOverflow and was kind of amazed that it actually worked. Go is really powerful in some ways that are actually a little bit scary. It is a really neat pattern though.